src/ui/common/src/components/workflows/artifact/metric/history.tsx

```diff
@@ -0,0 +1,80 @@
+import { CircularProgress } from '@mui/material';
+import Alert from '@mui/material/Alert';
+import Box from '@mui/material/Box';
+import React from 'react';
+import Plot from 'react-plotly.js';
+
+import PaginatedTable from '../../../../components/tables/PaginatedTable';
+import { ArtifactResultsWithLoadingStatus } from '../../../../reducers/artifactResults';
+import { Data, DataSchema } from '../../../../utils/data';
+import { isFailed, isInitial, isLoading } from '../../../../utils/shared';
+
+type Props = {
+  historyWithLoadingStatus?: ArtifactResultsWithLoadingStatus;
+};
+
+const metricHistorySchema: DataSchema = {
+  fields: [
+    { name: 'status', type: 'varchar' },
+    { name: 'timestamp', type: 'varchar' },
+    { name: 'value', type: 'float' },
+  ],
+  pandas_version: '',
+};
+
+const MetricsHistory: React.FC<Props> = ({ historyWithLoadingStatus }) => {
+  if (
+    !historyWithLoadingStatus ||
+    isInitial(historyWithLoadingStatus.status) ||
+    isLoading(historyWithLoadingStatus.status)
+  ) {
+    return <CircularProgress />;
+  }
+  if (isFailed(historyWithLoadingStatus.status)) {
+    return (
+      <Alert title="Failed to load historical data.">
+        {historyWithLoadingStatus.status.err}
+      </Alert>
+    );
+  }
+
+  const historicalData: Data = {
+    schema: metricHistorySchema,
+    data: (historyWithLoadingStatus.results?.results ?? []).map(
+      (artifactStatusResult) => {
+        return {
+          status: artifactStatusResult.exec_state?.status ?? 'Unknown',
+          timestamp: artifactStatusResult.exec_state?.timestamps?.finished_at,
+          value: artifactStatusResult.content_serialized,
+        };
+      }
+    ),
+  };
+
+  const dataToPlot = historicalData.data.filter(
+    (x) => !!x['timestamp'] && !!x['value']
+  );
+  const timestamps = dataToPlot.map((x) => x['timestamp']);
+  const values = dataToPlot.map((x) => x['value']);
+  return (
+    <Box display="flex" justifyContent="center" flexDirection="column">
+      <Plot
+        data={[
+          {
+            x: timestamps,
+            y: values,
+            type: 'scatter',
+            mode: 'lines+markers',
+            // colors are 'blue.900'. Plot doesn't seem to accept color from theme.
```

Comment:
@agiron123 I don't think plotly like `blue.900` so I put the hex here. lmk if there are better ways to solve this

Reply:
Yeah, import the theme object and use template literals. This is how we normally deal with react components that do not have the sx prop or our theme context.

```
color: `${theme.blue900}`
```
or however it should be indexed.